lib/rspec/core/metadata_filter.rb

```diff
@@ -42,6 +43,14 @@ def filter_applies_to_any_value?(key, value, metadata)
           metadata[key].any? { |v| filter_applies?(key, v,  key => value) }
         end
 
+        def id_filter_applies?(rerun_paths_to_scoped_ids, metadata)
+          scoped_ids = rerun_paths_to_scoped_ids.fetch(metadata[:rerun_file_path]) { return false }
```

Comment:
It's also necessary here, for the same reasons -- I'm using `return` to early exit from `id_filter_applies?` with a value of `false`.  Clearly `scoped_ids = false` doesn't make sense.